lib/Extractor/Solver.cpp

```diff
@@ -206,8 +206,24 @@ class BaseSolver : public Solver {
     }
 
     if (InferInsts && SMTSolver->supportsModels()) {
+      std::vector<Inst *> LHSComps;
+      findCands(LHS, LHSComps, IC, MaxNops);
+
+      std::vector<Inst *> Copies;
+      std::vector<InstMapping> PCsCopy;
+      BlockPCs BPCsCopy;
+      for (auto I : LHSComps) {
+        // separate sub-expressions by copying vars
+        std::map<Inst *, Inst *> InstCache;
+        std::map<Block *, Block *> BlockCache;
+        Copies.push_back(getInstCopy(I, IC, InstCache, BlockCache));
+        separateBlockPCs(BPCs, BPCsCopy, InstCache, BlockCache, IC);
+        separatePCs(PCs, PCsCopy, InstCache, BlockCache, IC);
```

Comment:
The synthesis engine doesn't take PCs/BPCs into account that are attached to the component library (RHS). Now, if a candidate from LHS has a PC/PBC, I should consider this during synthesis. This makes things a bit more complex. Maybe we can start without PC/BPC support first.

Reply:
Yes, just ignore for now.

I *think* the correct thing to do with pc/blockpc is to ignore them for almost all purposes (cost functions etc.) but to allow them to put constraints on the LHS, which should make synthesis work more often.